src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,68 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { setCurrentTodo } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(currentTodo?.userId || 0)
```

Comment:
Ensure that the `getUser` function can handle cases where `currentTodo?.userId` might be `0`, as this could lead to unexpected behavior.